wsit/ws-sx/wssx-impl/src/main/java/com/sun/xml/wss/provider/wsit/JMACAuthConfigFactory.java

```diff
@@ -235,24 +240,71 @@ public JMACAuthConfigFactory(ClassLoader loader) {
      *		permission to register a provider at the factory.
      */
     @Override
-    @SuppressWarnings("unchecked")
     public String registerConfigProvider(String className,
-					 Map properties, 
-					 String layer, String appContext, 
-					 String description) { 
+					 Map<String, String> properties,
+					 String layer, String appContext,
+					 String description) {
         //XXX do we need doPrivilege here
-        AuthConfigProvider provider =
-            _constructProvider(className, properties, null);
-        return _register(provider,properties,
-            layer,appContext,description,true);
+        AuthConfigProvider provider = _constructProvider(className, properties, null);
+        return _register(provider, properties, layer, appContext, description, true);
     }
 
     @Override
     public String registerConfigProvider(AuthConfigProvider provider,
-                                         String layer, String appContext, String description) {
-	return _register(provider,null,layer,appContext,description,false);
+        String layer, String appContext, String description) {
+        return _register(provider, null, layer, appContext, description, false);
+    }
+
+
+    @Override
+    public String registerServerAuthModule(ServerAuthModule serverAuthModule, Object context) {
```

Comment:
These two methods should be carefully reviewed; there is no test for them. The implementation is inspired by something similar what @arjantijms  made in Soteria, but customized for WSIT the case.